de.dlr.sc.virsat.model.extension.requirements/src/de/dlr/sc/virsat/model/extension/requirements/csv/CsvRequirementsImporter.java

```diff
@@ -0,0 +1,287 @@
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+
+import org.eclipse.emf.common.command.CompoundCommand;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.edit.domain.EditingDomain;
+
+import de.dlr.sc.virsat.model.concept.list.IBeanList;
+import de.dlr.sc.virsat.model.dvlm.Repository;
+import de.dlr.sc.virsat.model.dvlm.concepts.Concept;
+import de.dlr.sc.virsat.model.dvlm.concepts.util.ActiveConceptHelper;
+import de.dlr.sc.virsat.model.extension.requirements.Activator;
+import de.dlr.sc.virsat.model.extension.requirements.model.AttributeValue;
+import de.dlr.sc.virsat.model.extension.requirements.model.EnumerationLiteral;
+import de.dlr.sc.virsat.model.extension.requirements.model.Requirement;
+import de.dlr.sc.virsat.model.extension.requirements.model.RequirementAttribute;
+import de.dlr.sc.virsat.model.extension.requirements.model.RequirementObject;
+import de.dlr.sc.virsat.model.extension.requirements.model.RequirementType;
+import de.dlr.sc.virsat.model.extension.requirements.model.RequirementsConfiguration;
+import de.dlr.sc.virsat.project.resources.VirSatResourceSet;
+
+/**
+ * A class that imports requirements from a CSV file
+ *
+ */
+public class CsvRequirementsImporter {
+
+	protected EditingDomain editingDomain;
+	protected Concept reqConcept;
+
+	private static final String REQ_TYPE_NAME = "CCVImportedRequirementType";
+	private static final String REQ_EXTENSION_IDENTIFIER_PREFIX = "-extension";
+
+	/**
+	 * The function to load requirement contents from a given SCV file to an exiting
+	 * requirement specification
+	 * 
+	 * @param editingDomain
+	 *            the editing domain in which the import should be performed
+	 * @param csvContentMatrix
+	 *            the CSV content matrix
+	 * @param targetSpecificationList
+	 *            the specification list where the requirements should be imported
+	 *            into
+	 * @param attributeMapping
+	 *            the mapping of column index to attribute
+	 * @param importType
+	 *            the requirement type of the imported requirements
+	 * @return the compound command that can be performed to import the requirements
+	 */
+	public CompoundCommand loadRequirements(EditingDomain editingDomain, List<List<String>> csvContentMatrix,
+			IBeanList<RequirementObject> targetSpecificationList, Map<Integer, RequirementAttribute> attributeMapping,
+			RequirementType importType) {
+
+		this.editingDomain = editingDomain;
+		if (reqConcept == null) {
+			this.reqConcept = getReqConcept(importType.getTypeInstance());
+		}
+
+		CompoundCommand importCommand = new CompoundCommand();
+
+		// Read the actual requirements
+		for (List<String> req : csvContentMatrix) {
+			int lineNumber = csvContentMatrix.indexOf(req);
+			Requirement newReqElement = createRequirement(importType, lineNumber);
+
+			//If the first columns of a requirement are empty then values from previous lines are used
+			String attribute = req.iterator().next();

```

Comment:
Maybe `req.get(0)` would make it clearer that we are talking about the first column here.

Reply:
We're not only talking about the first column here... But all columns (starting from left) that do not have a value